packages/extension-commerce-bm-seo/src/components/seo-hoc.tsx

```diff
@@ -29,18 +29,43 @@ const getComponent = (
     return ComponentClass
 }
 
+const CATCH_ALL_PATH = '*'
+
+/**
+ * Checks whether the given URL path matches a predefined route defined in the application's routes config, excluding the catch-all route (e.g., path='*')
+ */
+const isRouteDefined = (routeToMatch: string, routes: Array<{path: string}>): boolean => {
+    // Exclude any catch-all (404) routes
+    const validRoutes = routes.filter((route) => route.path !== CATCH_ALL_PATH)
+
+    const isMatch = validRoutes.some(({path}) => {
+        return (
+            matchPath(routeToMatch, {
+                path,
+                exact: true
+            }) !== null
+        )
+    })
+    return isMatch
+}
+
 const seoHOC = <P extends object>(WrappedComponent: React.ComponentType<P>) => {
     const SeoHOC: React.FC<P> = (props: SeoHOCProps) => {
         const location = useLocation()
         const {routes, setRoutes} = useRoutes()
-        const {resourceTypeToComponentMap} = useExtensionConfig()
+        const {resourceTypeToComponentMap, matchingStrategy} = useExtensionConfig()
         const [urlSegment, setUrlSegment] = useState(location.pathname)
         const {setIsNavigationBlocked, siteLocale} = useApplicationExtensionsStore((state) => {
             return state.state['@salesforce/extension-commerce-bm-seo']
         })
-
         const resolveRef = useRef<(result?: object) => void>()
 
+        // The `matchingStrategy` configuration determines whether we check the CACHE (AKA the predefined route config) first or the `getUrlMapping` API
+        // `matchingStrategy == CACHE_FIRST`: if `location.pathname` matches a predefined route, skip the `getUrlMapping` API call
+        // `matchingStrategy == API_FIRST`: always call `getUrlMapping`
+        const skipMappingCall =
+            matchingStrategy === 'CACHE_FIRST' && isRouteDefined(location.pathname, routes)
```

Comment:
We can probably go ahead and create an enum and a type for the matching strategy configuration so we aren't comparing with a static string here.